cpp/src/strings/copying/copy_range.cu

```diff
@@ -0,0 +1,137 @@
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/detail/offsets_iterator_factory.cuh>
+#include <cudf/detail/valid_if.cuh>
+#include <cudf/strings/detail/copy_range.hpp>
+#include <cudf/strings/detail/strings_children.cuh>
+#include <cudf/strings/string_view.cuh>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/types.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <thrust/for_each.h>
+#include <thrust/iterator/counting_iterator.h>
+
+namespace cudf {
+namespace strings {
+namespace detail {
+namespace {
+struct compute_element_size {
+  column_device_view d_source;
+  column_device_view d_target;
+  size_type source_begin;
+  size_type target_begin;
+  size_type target_end;
+  bool source_has_nulls;
+  bool target_has_nulls;
+
+  __device__ cudf::size_type operator()(cudf::size_type idx)
+  {
+    if (idx >= target_begin && idx < target_end) {
+      auto const str_idx = source_begin + (idx - target_begin);
+      return source_has_nulls && d_source.is_null_nocheck(str_idx)
+               ? 0
+               : d_source.element<string_view>(str_idx).size_bytes();
+    } else {
+      return target_has_nulls && d_target.is_null_nocheck(idx)
+               ? 0
+               : d_target.element<string_view>(idx).size_bytes();
+    }
+  }
+};
+
+}  // namespace
+
+std::unique_ptr<column> copy_range(strings_column_view const& source,
+                                   strings_column_view const& target,
+                                   size_type source_begin,
+                                   size_type source_end,
+                                   size_type target_begin,
+                                   rmm::cuda_stream_view stream,
+                                   rmm::mr::device_memory_resource* mr)
+{
+  auto target_end = target_begin + (source_end - source_begin);
+  CUDF_EXPECTS(
+    (target_begin >= 0) && (target_begin < target.size()) && (target_end <= target.size()),
+    "Range is out of bounds.");
+
+  if (target_end == target_begin) { return std::make_unique<column>(target.parent(), stream, mr); }
+  auto source_device_view = column_device_view::create(source.parent(), stream);
+  auto d_source           = *source_device_view;
+  auto target_device_view = column_device_view::create(target.parent(), stream);
+  auto d_target           = *target_device_view;
+
+  // create null mask
+  auto [null_mask, null_count] = cudf::detail::valid_if(
+    thrust::make_counting_iterator<size_type>(0),
+    thrust::make_counting_iterator<size_type>(target.size()),
+    [d_source, d_target, source_begin, target_begin, target_end] __device__(size_type idx) {
+      return (idx >= target_begin && idx < target_end)
+               ? d_source.is_valid(source_begin + (idx - target_begin))
+               : d_target.is_valid(idx);
+    },
+    stream,
+    mr);
+
+  auto [check_source, check_target] = [target, null_count = null_count] {
```

Comment:
```suggestion
  auto [check_source, check_target] = [target, null_count] {
```

Reply:
This is necessary because `null_count` is created as a binding variable and cannot be passed directly in the capture area.